src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/ExportControllerTests.cs

```diff
@@ -112,38 +127,102 @@ public async Task GivenAnExportResourceTypeIdRequest_WhenResourceTypeIsNotGroup_
 
-            await Assert.ThrowsAsync<RequestNotValidException>(() => _exportEnabledController.ExportResourceTypeById(
+            await Assert.ThrowsAsync<RequestNotValidException>(() => exportController.ExportResourceType(
                 typeFilter: null,
                 since: null,
                 resourceType: null,
-                containerName: null,
+                containerName: _testContainer,
                 formatName: null,
+                anonymizationConfigCollectionReference: null,
                 anonymizationConfigLocation: null,
-                anonymizationConfigFileETag: anonymizationConfigEtag,
-                typeParameter: ResourceType.Patient.ToString(),
-                idParameter: "id"));
+                anonymizationConfigFileETag: _testAnonymizationConfigEtag,
+                typeParameter: ResourceType.Patient.ToString()));
+        }
+
+        [Fact]
+        public async Task GivenAnAnonymizedExportRequestWithAnonymizationConfigCollectionReference_WhenNoAnonymizationConfig_ThenRequestNotValidExceptionShouldBeThrown()
+        {
+            var exportController = GetController(_exportEnabledJobConfiguration, _anonymizationEnabledFeatureConfiguration, _artifactStoreConfig);
+
+            await Assert.ThrowsAsync<RequestNotValidException>(() => exportController.ExportResourceType(
+                typeFilter: null,
+                since: null,
+                resourceType: null,
+                containerName: _testContainer,
+                formatName: null,
+                anonymizationConfigCollectionReference: _testAnonymizationConfigCollectionReference,
+                anonymizationConfigLocation: null,
+                anonymizationConfigFileETag: null,
+                typeParameter: ResourceType.Patient.ToString()));
         }
 
-        private ExportController GetController(ExportJobConfiguration exportConfig)
+        [Fact]
+        public async Task GivenAnAnonymizedExportRequestWithAnonymizationConfigCollectionReference_WhenHasAnonymizationConfigEtag_ThenRequestNotValidExceptionShouldBeThrown()
+        {
+            var exportController = GetController(_exportEnabledJobConfiguration, _anonymizationEnabledFeatureConfiguration, _artifactStoreConfig);
+
+            await Assert.ThrowsAsync<RequestNotValidException>(() => exportController.ExportResourceType(
+                typeFilter: null,
+                since: null,
+                resourceType: null,
+                containerName: _testContainer,
+                formatName: null,
+                anonymizationConfigCollectionReference: _testAnonymizationConfigCollectionReference,
+                anonymizationConfigLocation: _testConfig,
+                anonymizationConfigFileETag: _testAnonymizationConfigEtag,
+                typeParameter: ResourceType.Patient.ToString()));
+        }
+
+        [Theory]
```

Comment:
Could you add a comment with the valid collection reference so that it is more clear what error each test case is testing?

Reply:
Sure~Added.